### ISSUE TITLE
{"error":{"code":500,"message":"com.esri.arcgis.portal.admin.core.site.migration.MigrationException: Failed to upgrade the PostgreSQL from 10.10 to 12.4.","details":null}}

### ISSUE THREAD

**Issue by jeffferdinand** [Esri/arcgis-powershell-dsc#339]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request

### Module Version

* 3.2

### Affected Resource(s)

* ArcGIS_Portal

### Configuration Files

```
# Copy-paste your DSC JSON configurations here - for large configs,
# please use a service like Dropbox and share a link to the ZIP file.
```

```JSON
{
  "AllNodes": [
    {
      "NodeName": "mygisserver.domain.com",
      "Role": [
        "Server",
        "Portal",
        "DataStore",
        "ServerWebAdaptor",
        "PortalWebAdaptor"
      ],
      "DataStoreTypes": [
        "Relational"
      ],
      "SslCertificates": [
        {
          "Path": "C:\\_GISServerBuildScripts\\ServerPrerequisites\\ssl_certificates\\gis-wildcard.lvvwd.com.pfx",
          "Password": "secret",
          "CNameFQDN": "doesntmatter",
		  "Alias": "doesntmatter",
          "Target": [
            "Server",
			"Portal",
			"DataStore",
			"WebAdaptor"
          ]
        }
      ],
	  "PortalLicenseFilePath": "C:\\_GISServerBuildScripts\\ArcGISEnterprise\\license_files\\10.9\\dev-ArcGIS_Enterprise_Portal_109_356561_20210722.json",
	  "ServerLicenseFilePath": "C:\\_GISServerBuildScripts\\ArcGISEnterprise\\license_files\\10.9\\dev-ArcGISGISServerAdvanced_StagingArcGISServer_1048103.prvc"
    }
  ],
    "ConfigData": {
        "Version": "10.9",
		"OldVersion":"10.8.1", 
        "ServerContext": "arcgis",
        "PortalContext": "portal",
        "ServerRole": "GeneralPurposeServer",
        "Credentials": {
            "ServiceAccount": {
                "UserName": "user",
                "Password": "secret",
                "IsDomainAccount": true,
				"IsMSAAccount": false
            }
        },
        "Server": {
            "LicenseFilePath": "C:\\_GISServerBuildScripts\\ArcGISEnterprise\\license_files\\10.9\\dev-ArcGISGISServerAdvanced_StagingArcGISServer_1048103.prvc",
            "Installer": {
                "Path": "C:\\ArcGIS_Server_Windows_109_177775.exe",
                "InstallDir": "C:\\ArcGIS\\Server",
                "InstallDirPython": "C:\\Python27"
            },
            "ServerDirectoriesRootLocation": "E:\\arcgisserver\\directories",
            "ConfigStoreLocation": "E:\\arcgisserver\\config-store",
            "LocalRepositoryPath": "E:\\arcgisserver\\local",
            "ServerLogsLocation": "E:\\arcgisserver\\logs",
            "ServerDirectories": [
                {
                    "name": "arcgiscache",
                    "physicalPath": "E:\\arcgisserver\\directories\\arcgiscache",
                    "directoryType": "CACHE",
                    "cleanupMode": "NONE",
                    "maxFileAge": 0
                },
                {
                    "name": "arcgisjobs",
                    "physicalPath": "E:\\arcgisserver\\directories\\arcgisjobs",
                    "directoryType": "JOBS",
                    "cleanupMode": "TIME_ELAPSED_SINCE_LAST_MODIFIED",
                    "maxFileAge": 360
                },
                {
                    "name": "arcgisoutput",
                    "physicalPath": "E:\\arcgisserver\\directories\\arcgisoutput",
                    "directoryType": "OUTPUT",
                    "cleanupMode": "TIME_ELAPSED_SINCE_LAST_MODIFIED",
                    "maxFileAge": 10
                },
                {
                    "name": "arcgissystem",
                    "physicalPath": "E:\\arcgisserver\\directories\\arcgissystem",
                    "directoryType": "SYSTEM",
                    "cleanupMode": "NONE",
                    "maxFileAge": 0
                }
            ],
            "RegisteredDirectories": [
                {
                    "name": "BasemapsCacheDir",
                    "physicalPath": "E:\\arcgisserver\\additionaldirs\\cache",
                    "directoryType": "CACHE"
                },
                {
                    "name": "TestOutputDir",
                    "physicalPath": "E:\\arcgisserver\\additionaldirs\\testOutput",
                    "directoryType": "OUTPUT"
                },
                {
                    "name": "TestJobsDir",
                    "physicalPath": "E:\\arcgisserver\\additionaldirs\\testJobs",
                    "directoryType": "JOBS"
                }
            ],
            "PrimarySiteAdmin": {
                "UserName": "secret",
                "Password": "secret"
            }
			,"ExternalLoadBalancer": "myalias.domain.com"
			,"InternalLoadBalancer": "mygisserverfqdn.domain.com"
        },
        "Portal": {
            "LicenseFilePath": "C:\\_GISServerBuildScripts\\ArcGISEnterprise\\license_files\\10.9\\dev-ArcGIS_Enterprise_Portal_109_356561_20210722.json",
            "PortalLicenseUserTypeId": "creatorUT",
            "Installer": {
                "Path": "C:\\Portal_for_ArcGIS_Windows_109_177786.exe",
                "WebStylesPath": "C:\\Portal_for_ArcGIS_Web_Styles_Windows_109_177787.exe",
                "InstallDir": "C:\\ArcGIS\\Portal",
                "ContentDir": "E:\\portalforarcgis"
            },
            "ContentDirectoryLocation":"E:\\portalforarcgis\\content\\arcgisportal\\content",
            "EnableAutomaticAccountCreation": "true",
            "DefaultRoleForUser": "iAAAAAAAAAAAAAAA",
            "DefaultUserLicenseTypeIdForUser": "viewerUT",
            "PortalAdministrator": {
                "UserName": "secret",
                "Email": "blah@blah.com",
                "Password": "secret",
                "SecurityQuestionIndex": 2,
                "SecurityAnswer": "secret"
            }
			,"ExternalLoadBalancer": "myalias.domain.com"
			,"InternalLoadBalancer": "mygisserverfqdn.domain.com"
        },
        "DataStore": {
            "ContentDirectoryLocation": "E:\\arcgisdatastore",
            "EnableFailoverOnPrimaryStop": false,
            "Installer": {
                "Path": "C:\\ArcGIS_DataStore_Windows_109_177788.exe",
                "InstallDir": "C:\\ArcGIS\\DataStore"
            }
        },
        "WebAdaptor": {
            "AdminAccessEnabled": true,
            "Installer": {
                "Path": "C:\\ArcGIS_Web_Adaptor_for_Microsoft_IIS_109_177789.exe"
            }
        },
		"Federation":{
			"PortalHostName": "mygisserverfqdn.domain.com",
			"PortalPort": "7443",
			"PortalContext": "portal",
			"PortalAdministrator": {
                "UserName": "secret",
                "Password": "secret"
            }
		}
    }
}
```

### Expected Behavior

ArcGIS Enterprise should have upgraded from version 10.8.1 to version 10.9

### Actual Behavior

7/23/2021 4:23:51 PM: PowerShell DSC resource ArcGIS_PortalUpgrade  failed to execute Set-TargetResource functionality with error message: [ERROR]:- {"error":{"code":500,"message":"com.esri.arcgis.portal.admin.core.site.migration.MigrationException: Failed to upgrade the PostgreSQL from 10.10 to 12.4.","details":null}} 
7/23/2021 4:23:51 PM: The SendConfigurationApply function did not succeed.

### Steps to Reproduce

Invoke-ArcGISConfiguration -ConfigurationParametersFile "dsc-configs\gisservd_10.9_upgrade.json" -Mode Upgrade -DebugSwitch

### Important Factoids

Single Machine deployment
10.8.1 successfully installed with exact same json, arcgis DSC powershell version 3.2
10.9 was also successfully installed with this configuration. this issue is specific to upgrade.
windows server 2019
IIS
slashes in config file are actually double, this page is rendering them incorrectly.

**Comment by cameronkroeker** [comment]
Hello @jeffferdinand,

From what I can tell your json config file looks valid. The error is coming from portal, it failed to upgrade its internal postgres database (the database upgrade happens during the "**Continue Portal Upgrade**" process after the setup has finished installation). 

The ArcGIS Module will first extract the Portal_for_ArcGIS_Windows_109_177786.exe, and then execute the setup.exe, which performs a silent installation ("inplace" upgrade"). Once the setup is complete, portal will now be in "upgrade" mode. Then the ArcGIS module will send a POST request to `https://mygisserver.domain.com:7443/arcgis/portaladmin/upgrade` to pass in the authorization file to complete the "**Continue Portal Upgrade**" process. This is when the content and database get upgraded, and is where I suspect the upgrade is failing, essentially step 9:

https://enterprise.arcgis.com/en/portal/latest/administer/windows/upgrade-portal-for-arcgis.htm#ESRI_SECTION1_9562E73D22D64660BF789AF14577EC77

Therefore, I recommend reviewing\gathering the portal logs (in E:\portalforarcgis) to see if there is any helpful information as to what could have potentially caused the PostgresSQL database upgrade to fail. 

For example, you will want to search the portal logs for the key words, "Failed to upgrade the PostgreSQL from 10.10 to 12.4", and/or filter by the time stamp 7/23/2021 4:23:51.

You may also want to submit these logs to Esri Technical Support, as they can also help review the logs, and investigate the issue further by taking at look at the environment via a screen share.

Thanks,
Cameron K.

**Comment by jeffferdinand** [comment]
Really appreciate the answer. thanks for the help.  I will update this thread after try your suggestions

**Comment by jeffferdinand** [comment]
Just to keep the thread updated. We are experiencing the exact same error when upgrading manually. I would bet big money its something to do with our environment or setup. Working with support.

**Comment by cameronkroeker** [comment]
> Just to keep the thread updated. We are experiencing the exact same error when upgrading manually. I would bet big money its something to do with our environment or setup. Working with support.

Thank you @jeffferdinand for providing an update. I hope with the help of support you guys are able to find the culprit and resolve it soon! Good luck.

**Comment by cameronkroeker** [comment]
Hi @jeffferdinand, 

I just wanted to check in to see if by chance there was any progress made on this issue?

Thanks,
Cameron K.

**Comment by jeffferdinand** [comment]
Ya, just yesterday someone at ESRI suggested to me to update the site administrator password, then retry the upgrade. This did resolve the problem. 

There is a file <portalhome>\db\.config-store-connection.json that contains an encrypted copy of the site admin password. Upon resetting the site admin password to the same password, I noticed this file was updated with a new encrypted password string.  I am supposing that since the site I was upgrading was brought over to this server with the disaster recover tools, perhaps the encrypted string in that file was not valid on the new server.

In any case, I did successfully upgrade using the powershell tools.
